### PR TITLE
chore: Add operator status condition metrics

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

Add `operator_status_condition_count` and `operator_status_condition_transition_seconds` from operatorpkg for NodePool and NodeClaim

**How was this change tested?**

`make presubmit`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: hack/docgen.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ AWS_SDK_GO_PROMETHEUS_DIR=$(go list -m -f '{{ .Dir }}' github.com/jonathan-innis
 OPERATORPKG_DIR=$(go list -m -f '{{ .Dir }}' github.com/awslabs/operatorpkg)
 
 compatibilitymatrix
-go run hack/docs/metrics_gen_docs.go pkg/ "${KARPENTER_CORE_DIR}/pkg" "${CONTROLLER_RUNTIME_DIR}/pkg" "${AWS_SDK_GO_PROMETHEUS_DIR}" "${OPERATORPKG_DIR}/metrics" website/content/en/preview/reference/metrics.md
+go run hack/docs/metrics_gen_docs.go pkg/ "${KARPENTER_CORE_DIR}/pkg" "${CONTROLLER_RUNTIME_DIR}/pkg" "${AWS_SDK_GO_PROMETHEUS_DIR}" "${OPERATORPKG_DIR}" website/content/en/preview/reference/metrics.md
```
Comment (njtran) [inline hack/docgen.sh:16]: was it intentional to remove the /metrics path? seems like that's still the path where metrics are defined, unless you're going one level up so you can recursively get them from all other packages?

**Review by njtran — APPROVED** [review]
one q

File: hack/docgen.sh (hunk 1 above)
Comment (jonathan-innis) [reply hack/docgen.sh]: No, there's other metrics that are defined in the `status` package

**Review by rschalo — APPROVED** [review]
/lgtm

**Pull request merged**